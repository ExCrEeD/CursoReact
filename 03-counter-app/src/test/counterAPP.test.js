import {shallow} from 'enzyme'
import CounterApp from '../CounterAPP';


describe('Pruebas counter app', () => {
    let wrapper = shallow(<CounterApp/>);

    beforeEach(()=>{
         wrapper = shallow(<CounterApp/>);
    });

    test('prueba render componente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('prueba valor por defecto componente ', () => {
        const valor = 100;
        const wrapper = shallow(<CounterApp value={valor}/>)
        const valorContador = parseInt(wrapper.find('h2').text());

        expect(valor).toBe(valorContador);
    });

    test('debe incrementar contador con boton+1 ', () => {
        const valor = 10;
        wrapper.find('button').at(0).simulate('click');
        const valorContador = parseInt(wrapper.find('h2').text());
        expect(valor+1).toBe(valorContador);
    });

    test('debe incrementar contador con boton-1 ', () => {
        const valor = 10;
        wrapper.find('button').at(1).simulate('click');
        const valorContador = parseInt(wrapper.find('h2').text());
        expect(valor-1).toBe(valorContador);
    });

    test('boton reset establece valor por defecto ', () => {
        const valor = 105;
        const wrapper = shallow(<CounterApp value={valor}/>)
        wrapper.find('button').at(1).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        const valorContador = parseInt(wrapper.find('h2').text());
        expect(valor).toBe(valorContador);
    });


})
