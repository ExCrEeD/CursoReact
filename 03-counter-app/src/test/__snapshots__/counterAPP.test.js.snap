// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas counter app prueba render componente  1`] = `
<Fragment>
  <h1>
    CounterAPP
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
